### PR TITLE
EL-4237: Standalone Navigation: option to disable navigation links

### PR DESCRIPTION
**Pull request by JonathanToddMF**
#### Checklist

* [x] The contents of this PR are mine to submit. No third party code or other material is being committed.
* [x] New third party dependencies (if any) are linked via `package.json`. Third party dependencies are licensed under one of the following: Apache, MIT, BSD, Mozilla Public License, Eclipse Public License, or Oracle Binary Code License.
* [x] The code in this PR does not contain export-restricted encryption algorithms.

#### Ticket / Issue
https://portal.digitalsafe.net/browse/EL-4237

#### Description of Proposed Changes
1. Added disabled to navigation-item interface
2. Prevented activation of clicking and assigning of href when the nav-item is disabled
3. Added disabled class to list item which inherits the list disabled styling
4. Added new node to navigation testpage for testing disabled.
5. Added unit tests to cover disabled nodes routerLinks and click events.
6. Updated screenshots to accommodate new node.
7. Updated previous unit tests to take into account the 4th nav node.

#### Documentation CI URL

https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_EL-4237-standalone-nav-disable-links

#### Downstream Build(s)

[ux-aspects-micro-focus](https://jenkins.swinfra.net/job/SEPG/job/caf/job/caf~ux-aspects-micro-focus~EL-4237-standalone-nav-disable-links~CI)
[quantum-ux-aspects](https://jenkins.swinfra.net/job/SEPG/job/caf/job/caf~quantum-ux-aspects~EL-4237-standalone-nav-disable-links~CI)
[quantum](https://jenkins.swinfra.net/job/SEPG/job/caf/job/caf~quantum~EL-4237-standalone-nav-disable-links~CI)

### CODE REVIEW

File: e2e/tests/components/navigation/navigation.e2e-spec.ts [hunk 1]
```diff
@@ -213,6 +227,48 @@ describe('Navigation Tests', () => {
         expect(await page.getPageContent()).toBe('Accounts Add');
     });
 
+    it('should disable node when disabled is to true', async () => {
+        const items = await page.getTopLevelItems();
+        await page.selectItem(items[3]);
+        const children = await page.getItemChildren(items[3]);
+
+        expect(await page.isItemDisabled(children[0])).toBeTruthy();
```
Comment (donna-marijne) [inline e2e/tests/components/navigation/navigation.e2e-spec.ts:235]:
```suggestion
        expect(await page.isItemDisabled(children[0])).toBe(true);
```
It doesn't matter in this context, but as a rule I recommend that you never use "fuzzy logic" in automated tests. Assert the exact value that you expect.

File: e2e/tests/components/navigation/navigation.e2e-spec.ts [hunk 2]
```diff
@@ -213,6 +227,48 @@ describe('Navigation Tests', () => {
...
+        expect(await page.isItemDisabled(items[3])).toBeTruthy();
+
+        expect(await imageCompare('navigation-options-disabled')).toEqual(0);
+    });
+
+    it('should prevent routerLinks from activating when disabled is true', async () => {
+        const items = await page.getTopLevelItems();
+        await page.disableOptions.click();
+
+        browser.actions().mouseMove(items[3]).click();
```
Comment (donna-marijne) [inline e2e/tests/components/navigation/navigation.e2e-spec.ts:251]: This is async, so you probably need to `await`.

**Review by donna-marijne — CHANGES_REQUESTED** [review]
Thanks for the changes. I noticed a couple of issues:

1. It's using the 'no-drop' cursor, which we decided to stop using for disabled items.
2. The expander icon does not share the disabled appearance.
    image
3. Disabled items can receive focus and expand using the keyboard.
    [Image from Gyazo](https://gyazo.com/800348db30729e935f77ae6f76608a19)

https://plnkr.co/edit/yusqSJVqQdi8Noh7

A general tip for e2e tests, maybe you do this already: after writing them, temporarily back out your component fix and make sure that all of the new tests fail.

**Review by donna-marijne — CHANGES_REQUESTED** [review]
Thanks, there should also be test coverage for disabled items in the "tree" mode.

Any time you fix an issue, including during code review, you should consider adding test coverage for it.

**Comment by JonathanToddMF** [comment]
Thanks for the feedback Donna.

I have added tests to cover tree mode, focus and child nodes.

I also removed the 2 duplicated screenshots and added additional screenshot coverage of tree mode, focus and child nodes

**Comment by JonathanToddMF** [comment]
Waiting on build passing before re-requesting review

**Comment by donna-marijne** [comment]
Great, thank you, this test coverage is good.

**Review by donna-marijne — APPROVED** [review]

**Pull request merged**